### ISSUE TITLE
Toggle buttons

### ISSUE THREAD

**Issue by hakenr** [havit/Havit.Blazor#887]
There is a Bootstrap feature we haven't covered: https://getbootstrap.com/docs/5.3/forms/checks-radios/#toggle-buttons  
There is a pull request #879 from @mmonteagudo that proposes an implementation.

There are still questions to resolve:

1. How does this coexist with [Switches](https://getbootstrap.com/docs/5.3/forms/checks-radios/#switches), which we implemented using the widely-used `HxSwitch` component (derived from `HxCheckbox`)? Should the toggle-button be another derivative of `HxCheckbox`, or are checkbox/switch/toggle-button variants of the basic `HxCheckbox` component that should be selectable via a `Variant=CheckboxVariant.Regular|Switch|ToggleButton` parameter? The enum-based variant would likely allow us to configure `HxCheckboxList` to handle rendering with a single parameter.

2. How do we handle the optional [_button group_](https://getbootstrap.com/docs/5.3/components/button-group/)?

3. There are also [Button plugin on/off buttons](https://getbootstrap.com/docs/5.3/components/buttons/#button-plugin) with similar behavior. Should we consider these as a separate feature that doesn't affect our check/radio boxes implementation, or is there potential overlap (e.g., `HxCheckboxVariant.ButtonPlugin`)?

**Comment by crdo** [comment]
@hakenr it actually is there, but the implementation or use is somehow awkward as it requires use of pure Blazor InputCheckbox  with HxButtonGroup.

https://havit.blazor.eu/components/HxButtonGroup#checkboxes-and-radios

**Comment by mmonteagudo** [comment]
Yes, the intention here is to add the feature in the very handy CheckboxList and RadiobuttonList components for ease of use. I'm already using it in our application.

**Comment by hakenr** [comment]
release 4.21.0-pre01
